interpreter/interpreter_test.go

```diff
@@ -357,10 +373,19 @@ func TestEval(t *testing.T) {
 
 			ctx := dependenciestest.Default().Inject(context.Background())
 			sideEffects, _, err := runtime.Eval(ctx, src)
-			if !tc.wantErr && err != nil {
-				t.Fatal(err)
-			} else if tc.wantErr && err == nil {
-				t.Fatal("expected error")
+			if err != nil {
+				if !tc.wantErr {
+					t.Fatal(err)
+				}
+
+				// We expect an error, so it should be a non-internal Flux error.
+				fluxErr, ok := err.(*errors.Error)
+				if !ok {
+					t.Fatalf("expected error of type %T, got %T", fluxErr, err)
+				} else if fluxErr.Code == codes.Internal {
+					t.Fatalf("expected error to have code other than %s", codes.Internal)
+				}
+				return
```

Comment:
Do we need to check that we get the right error text? Not just the type?

Reply:
I guess that this test doesn't do that. There *are* other tests that *do* check error messages.  I think there is a balance where we want tests to be meaningful, but not so fragile what we have to update them often.

I considered adding logic to verify the error message, but there are several other test cases that are also expected to error, so I opted to just follow the convention in place for this test.

Reply:
Okay, that makes sense to me. Thanks for clarifying 😊